core/src/main/java/org/geysermc/geyser/translator/protocol/java/level/JavaLevelChunkWithLightTranslator.java

```diff
@@ -171,7 +193,8 @@ public void translate(GeyserSession session, ClientboundLevelChunkWithLightPacke
                     continue;
                 }
 
-                IntList bedrockPalette = new IntArrayList(javaPalette.size());
+                IntList bedrockPalette = new IntArrayList();
```

Comment:
Would we be able to add back the optimization of setting the size here in the instance no extended collisions are required in this chunk?

Reply:
https://github.com/GeyserMC/Geyser/pull/3505/commits/410712cb78f1e7df00d9099b409664714fafec16#diff-be261e57e1fd314463d26d271b48e1b9d0d74b67e755b097f44c9037da7a4627L196-R217